### PR TITLE
File replacement action

### PR DESCRIPTION
**Pull request by flamerohr**
This enables the user to upload a new file to replace the currently opened file in the asset-admin area.

Only supports uploading one file as a replacement.

Please merge https://github.com/silverstripe/silverstripe-framework/pull/6428 first.
Please see https://github.com/silverstripe/silverstripe-asset-admin/issues/295 for more details

### CODE REVIEW

File: code/Controller/AssetAdmin.php [hunk 1]
```diff
@@ -447,7 +455,74 @@ public function apiCreateFile(HTTPRequest $request)
...
     }
-
+    
+    /**
+     * Creates a single file based on a form-urlencoded upload.
+     *
+     * @param HTTPRequest $request
+     * @return HTTPRequest|HTTPResponse
+     */
+    public function apiUploadFile(HTTPRequest $request)
```
Comment (tractorcow) [inline code/Controller/AssetAdmin.php]: Too much duplication between this and apiCreateFile, with lack of docs distinguishing the different use cases.

Perhaps it should be apiReplaceFile to make the purpose clear?

Comment (tractorcow) [inline code/Controller/AssetAdmin.php]: Pushed a change to just document it better.

File: client/src/components/AssetDropzone/AssetDropzone.js [hunk 2]
```diff
@@ -326,9 +355,9 @@ class AssetDropzone extends SilverStripeComponent {
         category: this.getFileCategory(file.type),
         filename: file.name,
         queuedId: file._queuedId,
-        size: fileSize(file.size),
+        size: file.size,
```
Comment (tractorcow) [inline client/src/components/AssetDropzone/AssetDropzone.js]: Why did you remove this? We lose all formatting. e.g. a `1.8 MB` file will appear as `1800000` now.

image

Comment (flamerohr) [inline client/src/components/AssetDropzone/AssetDropzone.js]: Removed because it was causing double formatting.
The reverse is that you get "NaN" when uploading in the table view.

Comment (tractorcow) [inline client/src/components/AssetDropzone/AssetDropzone.js]: I've fixed this by adding fileSize to UploadFieldItem.js

**Comment by flamerohr** [comment]
Thanks for the fixes @tractorcow, they all look good!
